File: pedro_vitor_problema1.py
import math

class complexo:

    def __init__(self, initX, initY, initFormato):

        self.x = float(initX)
        self.y = float(initY)
        self.formato = initFormato
 
    def convert(point1, FS):
        
        if point1.formato != FS :
            if FS == 'E':
                point1.x =  ((float(point1.x) ** 2) + (float(point1.y) ** 2)) ** 0.5
                point1.y =  math.atan(float(point1.y)/float(point1.x))
                
            elif FS == 'C':
                point1.x = float(point1.x) * math.cos(float(point1.y))
                point1.y = float(point1.x) * math.sin(float(point1.y))
                
        elif point1.formato == FS :
            point1.x = point1.x
            point1.y = point1.y
            
    def complexOp (self, p1, p2, operacao, FS):
            
       if operacao == 'M' :
           p1.convert('E')
           p2.convert('E')
           R.x = p1.x*p2.x
           R.y = p1.y+p2.y
           R.formato = 'E'
       elif operacao == 'D' :  
           p1.convert('E')
           p2.convert('E')
           R.x = p1.x/p2.x
           R.y = p1.y-p2.y
           R.formato = 'E'
       elif operacao == 'A' :  
           p1.convert('C')
           p2.convert('C')
           R.x = p1.x+p2.x
           R.y = p1.y+p2.y
           R.formato = 'C'
       elif operacao == 'S' :  
           p1.convert('C')
           p2.convert('C')
           R.x = p1.x-p2.x
           R.y = p1.y-p2.y
           R.formato = 'C'
                
       if FS == 'C' :
           R.convert('C')
       elif FS == 'E':
           R.convert('E')
        
A = complexo(0,0,'C')
B = complexo(0,0,'C')
R = complexo(0,0,'C')

print("Calculadora de números complexos:")
print("*********************************")
print("Escolha uma opção:")
print(" C => Conversão de coordenadas")
print(" O => Operação entre numeros complexos")
print(" Z => Sair do programa")
opcao = input()

while opcao != "Z":
    
    if opcao == "O":

        A.formato = input("Sistema de coordenadas do primeiro numero:")
        
        if A.formato == 'C':
            A.x = float(input("Parte real do primeiro numero:"))
            A.y = float(input("Parte imaginaria do primeiro numero:"))
        
        elif A.formato == 'E' :
            A.x = float(input("Modulo do primeiro numero:"))
            A.y = float(input("Fase do primeiro numero:"))
        else: 
            print("Opção inválida.")
            
        B.formato = input("Sistema de coordenadas do segundo numero:")
        
        if B.formato == 'C':
            B.x = float(input("Parte real do segundo numero:"))
            B.y = float(input("Parte imaginaria do segundo numero:"))
        
        elif B.formato == 'E' :
            B.x = float(input("Modulo do segundo numero:"))
            B.y = float(input("Fase do segundo numero:"))
        else: 
            print("Opção inválida.")
        
        op = input("Digite a operação desejada:")  
        
        out = input("Digite o formato de saida desejado:")
        
        R.complexOp(A,B,op,out)
        
        if out == 'E':
            print("Modulo:")
            print(R.x)
            print("Fase:")
            print(R.y)
        elif out == 'C':
            print("Real:")
            print(R.x)
            print("Imaginária:")
            print(R.y)
            
        print("Escolha uma opção:")
        print(" C => Conversão de coordenadas")
        print(" O => Operação entre numeros complexos")
        print(" Z => Sair do programa")
        opcao = input()
            
    elif opcao == "C":
        
        A.formato = input("Sistema de coordenadas de seu numero:")
        
        if A.formato == 'C':
            A.x = float(input("Parte real de seu numero:"))
            A.y = float(input("Parte imaginaria de seu numero:"))
        
        elif A.formato == 'E' :
            A.x = float(input("Modulo de seu numero:"))
            A.y = float(input("Fase de seu numero:"))
        else: 
            print("Opção inválida.")
        
        out = input("Digite o formato de saida desejado:")
        
        A.convert(out)
        
        if out == 'E':
            print("Modulo:")
            print(A.x)
            print("Fase:")
            print(A.y)
        elif out == 'C':
            print("Real:")
            print(A.x)
            print("Imaginária:")
            print(A.y)
            
        print("Escolha uma opção:")
        print(" C => Conversão de coordenadas")
        print(" O => Operação entre numeros complexos")
        print(" Z => Sair do programa")
        opcao = input()
        

File: pedro_vitor_problema1.c
        #include <stdio.h>
        #include <stdlib.h>
        #include <math.h>
        #define PI 3.14159

        struct complexo{

            float x, y;
            char formato;

        };

        void convComplexo(struct complexo *, char); // Conversao

        void complexOp(char, struct complexo, struct complexo, struct complexo *, char); // Operacao


        int main ()
        {

            struct complexo A, B, R;
            char conversion, operacao, FS;
            char opcao;

            printf("*********************************\n");
            printf("Calculadora de numeros complexos:\n");
            printf("*********************************\n");
            printf("Elaborada por: Pedro Vitor Ferreira Goncalves\n");
            printf("*********************************\n");

            printf("Escolha uma opcao:\n");
            printf("1) C => Conversao de coordenadas\n2) O => Operacao entre numeros complexos\n3) Z => Fim do programa\n");
            scanf("%c", &opcao);
            printf("*********************************\n");

            while (opcao != 'Z')
            {

                if(opcao == 'O')
                {

                    getchar();

                    printf("Sistema de coordenadas de seu primeiro numero:\n C => Cartesiana E => Exponencial\n");
                    scanf("%c", &A.formato);

                    if( A.formato == 'C')
                    {
                        printf("Parte real de seu primeiro numero:\n");
                        scanf("%f", &A.x);
                        printf("Parte imaginaria de seu primeiro numero:\n");
                        scanf("%f", &A.y);
                    }
                    else if( A.formato == 'E')
                    {
                        printf("Modulo de seu primeiro numero:\n");
                        scanf("%f", &A.x);
                        printf("Fase de seu primeiro numero:\n");
                        scanf("%f", &A.y);
                    } else
                        printf("Opcao invalida.\n");

                    getchar();

                    printf("Sistema de coordenadas de seu segundo numero:\n C => Cartesiana E => Exponencial\n");
                    scanf("%c", &B.formato);

                    if( B.formato == 'C')
                    {
                        printf("Parte real de seu segundo numero:\n");
                        scanf("%f", &B.x);
                        printf("Parte imaginaria de seu segundo numero:\n");
                        scanf("%f", &B.y);
                    }
                    else if( B.formato == 'E')
                    {
                        printf("Modulo de seu segundo numero:\n");
                        scanf("%f", &B.x);
                        printf("Fase de seu segundo numero:\n");
                        scanf("%f", &B.y);
                    } else
                        printf("Opcao invalida.\n");

                    printf("Digite a operacao desejada:\n");
                    getchar();
                    scanf("%c", &operacao);

                    printf("Digite o formato de saida desejado:\n");
                    getchar();
                    scanf("%c", &FS);

                    complexOp(operacao, A, B, &R, FS);



                    if ( FS == 'E')
                    {
                            printf("\nModulo: %.2f\n", R.x);
                            printf("Fase: %.2f radianos\n", R.y);
                            printf("Fase: %.2f graus.\n", R.y * 180 / PI);
                            printf("Formato: %c\n", R.formato);
                    }
                    else if ( FS == 'C')
                    {
                            printf("\nParte real: %.2f\n", R.x);
                            printf("Parte imaginaria: %.2f\n", R.y);
                            printf("Formato: %c\n", R.formato);
                    }

                    printf("*********************************\n");
                    getchar();
                    printf("Escolha uma opcao:\n");
                    printf("1) C => Conversao de coordenadas\n2) O => Operacao entre numeros complexos\n3) Z => Fim do programa\n");
                    scanf("%c", &opcao);

                } else if(opcao == 'C')
                {
                    getchar();

                    printf("Sistema de coordenadas de seu numero:\n C => Cartesiana E => Exponencial\n");
                    scanf("%c", &R.formato);

                    if( R.formato == 'C')
                    {
                        printf("Parte real de seu primeiro numero:\n");
                        scanf("%f", &R.x);
                        printf("Parte imaginaria de seu primeiro numero:\n");
                        scanf("%f", &R.y);
                    }
                    else if( R.formato == 'E')
                    {
                        printf("Modulo de seu primeiro numero:\n");
                        scanf("%f", &R.x);
                        printf("Fase de seu primeiro numero:\n");
                        scanf("%f", &R.y);
                    } else
                    {
                        printf("Opcao invalida.\n");
                        exit(1);
                    }

                    printf("Digite o formato de saida desejado:\n");
                    getchar();
                    scanf("%c", &FS);

                    convComplexo(&R, FS);



                    if ( FS == 'E')
                    {
                            printf("\nModulo: %.2f\n", R.x);
                            printf("Fase: %.2f radianos\n", R.y);
                            printf("Fase: %.2f graus.\n", R.y * 180 / PI);
                            printf("Formato: %c\n", R.formato);
                    }
                    else if ( FS == 'C')
                    {
                            printf("\nParte real: %.2f\n", R.x);
                            printf("Parte imaginaria: %.2f\n", R.y);
                            printf("Formato: %c\n", R.formato);
                    }

                    printf("*********************************\n");
                    getchar();
                    printf("Escolha uma opcao:\n");
                    printf("1) C => Conversao de coordenadas\n2) O => Operacao entre numeros complexos\n3) Z => Fim do programa\n");
                    scanf("%c", &opcao);

                
                }
            
            }


            system("PAUSE");

            return 0;

        }

        void convComplexo(struct complexo *Z, char CONVERSAO){

            float a, b;
            char c;

            if( CONVERSAO != (*Z).formato){
                if ( CONVERSAO == 'C'){
                    a = (*Z).x*cos((*Z).y); // parte real
                    b = (*Z).x*sin((*Z).y); // parte imaginaria
                    c = 'C';
                }
                else if ( CONVERSAO == 'E'){
                    a = sqrt(pow((*Z).x, 2) + pow((*Z).y, 2)); // modulo
                    b = atan((*Z).y/(*Z).x); // fase
                    c = 'E';
                }
            }
            else
            {
                a = (*Z).x;
                b = (*Z).y;
                c = (*Z).formato;
            }

            (*Z).x = a;
            (*Z).y = b;
            (*Z).formato = c;


        }

        void complexOp(char operacao, struct complexo Za, struct complexo Zb, struct complexo *Y, char FS){

            if ( operacao == 'M'){
                convComplexo(&Za, 'E');
                convComplexo(&Zb, 'E');
                (*Y).x = Za.x * Zb.x;
                (*Y).y = Za.y + Zb.y;
                (*Y).formato = 'E';
            }
            else if ( operacao == 'D' ){
                convComplexo(&Za, 'E');
                convComplexo(&Zb, 'E');
                (*Y).x = Za.x / Zb.x;
                (*Y).y = Za.y - Zb.y;
                (*Y).formato = 'E';
            }
            else if ( operacao == 'A' ){
                convComplexo(&Za, 'C');
                convComplexo(&Zb, 'C');
                (*Y).x = Za.x + Zb.x;
                (*Y).y = Za.y + Zb.y;
                (*Y).formato = 'C';
            }
            else if ( operacao == 'S' ){
                convComplexo(&Za, 'C');
                convComplexo(&Zb, 'C');
                (*Y).x = Za.x - Zb.x;
                (*Y).y = Za.y - Zb.y;
                (*Y).formato = 'C';
            }

            // Zres.x = Y.x;
            // Zres.y = Y.y;
            // Zres.formato = Y.formato;

            if( FS == 'C' )
                convComplexo(Y, 'C');
            else if( FS == 'E' )
                convComplexo(Y, 'E');

        }
